src/redux/globalSearch/reducer.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Reducer for globalSearch
+ */
+import * as Actions from './actions';
+
+const initialState = {
+  locationSearch: {
+    data: [],
+    query: '',
+    isFetching: false,
+    isFetched: false,
+  },
+};
+
+// the location page reducer
+function locationSearch(state = initialState.locationSearch, action = {}) {
+  switch (action.type) {
+    case Actions.FETCH_LOCATION_SEARCH:
+      return {
+        ...state,
+        data: state.data,
```

Comment:
FYI this is unnecessary due to the `...state` above it. I think I did it in other reducers though. The benefit of explicitly writing is to call out that we are keeping the previous state for data, but I don't think it's necessary to do so since `...state` and the absence of `data` does it already.